src/main/java/net/earthcomputer/clientcommands/command/ReplyCommand.java

```diff
@@ -3,32 +3,53 @@
 import com.mojang.brigadier.Command;
 import com.mojang.brigadier.CommandDispatcher;
 import com.mojang.brigadier.exceptions.CommandSyntaxException;
-import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
 import com.mojang.brigadier.exceptions.Dynamic2CommandExceptionType;
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
+import net.earthcomputer.clientcommands.Configs;
 import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
 import net.minecraft.SharedConstants;
 import net.minecraft.network.chat.Component;
-import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-import static dev.xpple.clientarguments.arguments.CMessageArgument.*;
-import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import static dev.xpple.clientarguments.arguments.CMessageArgument.getMessage;
+import static dev.xpple.clientarguments.arguments.CMessageArgument.message;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.argument;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;
 
 public class ReplyCommand {
+    public static final float MAXIMUM_REPLY_DELAY_SECONDS = 10.0f;
+
     private static final SimpleCommandExceptionType NO_TARGET_FOUND_EXCEPTION = new SimpleCommandExceptionType(Component.translatable("commands.creply.noTargetFound"));
     private static final Dynamic2CommandExceptionType MESSAGE_TOO_LONG_EXCEPTION = new Dynamic2CommandExceptionType((a, b) -> Component.translatable("commands.creply.messageTooLong", a, b));
 
+    private static final List<ReplyCandidate> replyCandidates = new ArrayList<>();
+
     @Nullable
-    private static String mostRecentWhisper = null;
-    @Nullable
-    private static String currentTarget = null;
+    public static String getCurrentTarget() {
+        long now = System.currentTimeMillis();
 
-    public static void refreshCurrentTarget() {
-        currentTarget = mostRecentWhisper;
+        for (int i = 0; i < replyCandidates.size(); i++) {
+            ReplyCandidate candidate = replyCandidates.get(i);
+            if ((now - candidate.timestampMs) / 1_000.0f > MAXIMUM_REPLY_DELAY_SECONDS) {
+                replyCandidates.remove(i--);
+            }
+        }
+
+        for (int i = replyCandidates.size() - 1; i >= 0; i--) {
+            ReplyCandidate candidate = replyCandidates.get(i);
+            if ((now - candidate.timestampMs) / 1_000.0f >= Configs.minimumReplyDelaySeconds) {
```

Comment:
Nitpick but I would multiply the other side by 1000 to convert to milliseconds. It's a micro-optimization but multiplication is more efficient than division and if it doesn't hurt readability may as well do the multiplication. Same for the one above